doc/src/manual/variables-and-scoping.md

```diff
@@ -743,7 +743,7 @@ ERROR: invalid redefinition of constant x
 julia> const y = 1.0
 1.0
 
-julia> y = 2.0
+julia> const y = 2.0
 WARNING: redefinition of constant y. This may fail, cause incorrect answers, or produce other errors.
```

Comment:
Shouldn't we still expect this warning to be emitted when redefining constants, so this test change is not really necessary?

Reply:
See above, this is the one semantic change I kept, I made
```
const x = 2
x = 2
```
and error without `const` keyword. Same applies to `setglobal!` in general. It's no longer trivial to support in codegen because these are stored in different places now and it just didn't seem worth the extra effort only to rip it out again.

Reply:
okay, yes, upgrading a warning -> error here sounds fine to me, even though it necessitated changing the tests